WebAPIBookStore/WebAPIBookStore/Usecase/ProductUsecase.cs

```diff
@@ -128,5 +123,55 @@ public Output Delete(int id)
 
             return _useCaseOutput.Success(_mapper.Map<ProductDto>(product));
         }
+
+        public Output Sort(bool descending, SortValue sortValue)
+        {
+            if (descending)
+            {
+                switch (sortValue)
+                {
+                    case SortValue.Name:
+                        var product1 = _productRepository.SortDown("Name");
+                        return product1.Any() ? _useCaseOutput.Success(product1) : _useCaseOutput.NotFound("Not found any product");
+
+                    case SortValue.Price:
+                        var product2 = _productRepository.SortDown("Price");
+                        return product2.Any() ? _useCaseOutput.Success(product2) : _useCaseOutput.NotFound("Not found any product");
+
+                    case SortValue.PuslishYear:
+                        var product3 = _productRepository.SortDown("PuslishYear");
+                        return product3.Any() ? _useCaseOutput.Success(product3) : _useCaseOutput.NotFound("Not found any product");
+
+                    default:
+                        return _useCaseOutput.InternalServer("Input not true");
+                }
+            }
+            else
+            {
+                switch (sortValue)
+                {
+                    case SortValue.Name:
+                        var product1 = _productRepository.SortUp("Name");
+                        return product1.Any() ? _useCaseOutput.Success(product1) : _useCaseOutput.NotFound("Not found any product");
+
+                    case SortValue.Price:
+                        var product2 = _productRepository.SortUp("Price");
+                        return product2.Any() ? _useCaseOutput.Success(product2) : _useCaseOutput.NotFound("Not found any product");
+
+                    case SortValue.PuslishYear:
+                        var product3 = _productRepository.SortUp("PuslishYear");
+                        return product3.Any() ? _useCaseOutput.Success(product3) : _useCaseOutput.NotFound("Not found any product");
+
+                    default:
```

Comment:
em cần define 1 dạng enum hoặc 1 đống int để định nghĩa cho các tiêu chí sort, mình ko sort bằng cách truyền name

- sort up/down ko tách thành 2 hàm như thế trông rất xấu, tham khảo kiểu như này:
- OrderByValue(int byValue, bool? AscSort)
- if (AscSort.HasValue())
- .......

Reply:
em ko cần thiết phải check notfound, nó sort mà có tìm cái gì đâu mà not found nhỉ, ko có thì trả ra mảng rỗng thôi em